prow/repoowners/repoowners.go

```diff
@@ -152,6 +152,11 @@ func (c *Client) LoadRepoAliases(org, repo string) (RepoAliases, error) {
 			return nil, fmt.Errorf("failed to clone %s: %v", fullName, err)
 		}
 		defer gitRepo.Clean()
+		if base != "master" {
```

Comment:
Why do we need this `if`?

Reply:
When a repo is cloned, default_branch is checked out which most of the time is master. If we target a different branch, we need to actually check that branch out and then load repo owners from OWNERS files.

Reply:
Right, but even if you are already on a ref you can check out the ref and `git` will no-op?

Reply:
Right - updated